### PR TITLE
GraphQLEndpoint initializer now checks that types have id

### PR DESCRIPTION
**Pull request by WiktorKarlsson**
Necessary fix that makes sure no types are initialized in GraphQLEndpoint initializer unless they have the ID field. Removes initialized fields if they would lead to such a removed type.

### CODE REVIEW

File: src/main/java/se/liu/ida/hefquin/engine/federation/GraphQLEndpoint.java [hunk 1]
```diff
@@ -58,4 +58,9 @@ public interface GraphQLEndpoint extends FederationMember {
      * If the parameter names provided doesn't correspond to an object type or a GraphQLEntrypointType for the endpoint return null.
      */
     public GraphQLEntrypoint getEntrypoint(final String objectTypeName, final GraphQLEntrypointType fieldType);
+
+    /*
+     * Uses System.out to print information about the endpoint such as GraphQL object types and their fields.
+     */
+    public void printInformation();
```
Comment (scferrada) [inline src/main/java/se/liu/ida/hefquin/engine/federation/GraphQLEndpoint.java:65]: Is this method really necessary? Usually this kinds of systems do not use the `System.out.print` methods to log information. If you want this info for debugging or to generate files with, I would recommend to implement the `toString()` method instead.

File: src/main/java/se/liu/ida/hefquin/engine/wrappers/graphqlwrapper/impl/GraphQLEndpointInitializerImpl.java [hunk 2]
```diff
@@ -151,6 +152,67 @@ protected static GraphQLQuery getIntrospectionQuery(){
...
+
+        final Set<String> typesToRemove = new HashSet<>();
+
+        // Check if any types needs to be removed
+        for(final String type : typesAndFields.keySet()){
+            final Map<String,GraphQLField> fields = typesAndFields.get(type);
+
+            // Check if type has an id field
+            if(!fields.containsKey("id")){
+                System.out.println("Type: " + type + " is removed due to not having an id field!");
```
Comment (scferrada) [inline src/main/java/se/liu/ida/hefquin/engine/wrappers/graphqlwrapper/impl/GraphQLEndpointInitializerImpl.java:171]: Also here, are these prints really necessary or were useful for you while developing?

**Review by scferrada — CHANGES_REQUESTED** [review]
Hi Wiktor, here's my review. I don't think any of these print utilities are necessary, so please check that.

File: src/main/java/se/liu/ida/hefquin/engine/wrappers/graphqlwrapper/impl/GraphQLEndpointInitializerImpl.java (hunk 2 above)
Comment (WiktorKarlsson) [reply src/main/java/se/liu/ida/hefquin/engine/wrappers/graphqlwrapper/impl/GraphQLEndpointInitializerImpl.java]: They are not necessary. They are now removed in latest commit: [94c8ce5](https://github.com/LiUSemWeb/HeFQUIN/pull/237/commits/94c8ce59988204af9c00434a7f9acc14d5c8e053)

**Comment by scferrada** [comment]
Thanks, it looks good now.

**Comment by hartig** [comment]
@scferrada can I merge this PR now?

**Comment by scferrada** [comment]
Yes, please.

**Review by scferrada — APPROVED** [review]

**Pull request merged**